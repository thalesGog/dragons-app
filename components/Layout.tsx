import React, {ReactNode} from 'react'
import Head from 'next/head'
import Container from './Container'
import Navbar from './Navbar'
import Link from 'next/link'
import {signOut, useSession} from 'next-auth/client'

type Props = {
  children?: ReactNode
  title?: string
}

const Layout = ({children, title = 'Dragons App'}: Props): JSX.Element => {
  const [session] = useSession()

  const logout = (): void => {
    signOut({callbackUrl: '/'})
  }
  return (
    <>
      <Head>
        <title>{title ? `Dragons App - ${title}` : title}</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      {session && (
        <Navbar>
          <img src="/logo.png" alt="Dragon's Logo" />
          <ul>
            <Link href="/">
              <li>In√≠cio</li>
            </Link>
            <Link href="/dragon/create">
              <li>Criar</li>
            </Link>
          </ul>
          <span role="button" onClick={logout}>
            Sair
          </span>
        </Navbar>
      )}

      <Container>{children}</Container>
    </>
  )
}

export default Layout
