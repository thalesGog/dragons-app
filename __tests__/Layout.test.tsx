import React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import client, {Session} from 'next-auth/client'
import Layout from '../components/Layout'
jest.mock('next-auth/client')

test('Render Layout component', async () => {
  const mockSession: Session = {
    expires: '1',
    user: {email: 'a', name: 'Delta', image: 'c'},
  }

  ;(client.useSession as jest.Mock).mockReturnValueOnce([mockSession, false])
  render(<Layout />)
})
