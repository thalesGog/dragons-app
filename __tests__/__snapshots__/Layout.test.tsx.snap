// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Layout component 1`] = `
.c1 {
  max-width: 1024px;
  margin: 0 auto;
  text-align: center;
  padding-left: 8px;
  padding-right: 8px;
  padding-bottom: 80px;
}

.c0 {
  border-bottom: 1px solid #10162f;
  background-color: #fff;
  position: -webkit-sticky;
  position: sticky;
  width: 100%;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 4px;
}

.c0 ul {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 ul li {
  padding: 8px;
  cursor: pointer;
  text-transform: uppercase;
  font-weight: 600;
}

.c0 ul li:hover {
  color: #3a10e5;
}

.c0 span {
  cursor: pointer;
}

.c0 span:hover {
  color: #3a10e5;
}

.c0 img {
  padding-top: 8px;
}

<div>
  <nav
    class="c0"
  >
    <img
      alt="Dragon's Logo"
      src="/logo.png"
    />
    <ul>
      <li>
        In√≠cio
      </li>
      <li>
        Criar
      </li>
    </ul>
    <span
      role="button"
    >
      Sair
    </span>
  </nav>
  <main
    class="c1"
  />
</div>
`;
